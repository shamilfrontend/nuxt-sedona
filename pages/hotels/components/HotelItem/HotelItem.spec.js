import { shallowMount } from '@vue/test-utils'
import Component from './index'

describe('HotelItem', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Component, {
      propsData: {
        hotel: {
          imagePath: 'hotel.png',
          title: 'Hotel Name',
          price: '2000',
          starCount: 1,
          rating: 10
        }
      }
    })
  })

  test('mounts properly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('methods', () => {
    describe('starClasses', () => {
      test('should return className "hotel-item__stars_one" if value equal 1', () => {
        const expected = 'hotel-item__stars_one'
        const result = wrapper.vm.starClasses(1)
        expect(result).toEqual(expected)
      })
      test('should return className "hotel-item__stars_second" if value equal 1', () => {
        const expected = 'hotel-item__stars_second'
        const result = wrapper.vm.starClasses(2)
        expect(result).toEqual(expected)
      })
      test('should return className "hotel-item__stars_three" if value equal 1', () => {
        const expected = 'hotel-item__stars_three'
        const result = wrapper.vm.starClasses(3)
        expect(result).toEqual(expected)
      })
      test('should return className "hotel-item__stars_four" if value equal 1', () => {
        const expected = 'hotel-item__stars_four'
        const result = wrapper.vm.starClasses(4)
        expect(result).toEqual(expected)
      })
      test('should return className "hotel-item__stars_five" if value equal 5', () => {
        const expected = 'hotel-item__stars_five'
        const result = wrapper.vm.starClasses(5)
        expect(result).toEqual(expected)
      })
      test('should return className "hotel-item__stars_one" if value equal any', () => {
        const expected = 'hotel-item__stars_one'
        const result = wrapper.vm.starClasses('any')
        expect(result).toEqual(expected)
      })
    })
  })
})
