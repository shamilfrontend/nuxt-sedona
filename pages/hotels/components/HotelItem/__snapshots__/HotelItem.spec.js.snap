// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelItem should match snapshot 1`] = `
<article class="hotel-item">
  <div class="hotel-item__left">
    <div class="hotel-item__image"><img src="hotel.png" alt="Amara Resost &amp; SPA" height="90" width="135"></div>
    <div class="hotel-item__content">
      <h3 class="hotel-item__name">Hotel Name</h3>
      <div class="hotel-item-type"><span class="hotel-item-type__name">Гостиница</span> <span class="hotel-item-type__price">От 2000 Р.</span></div> <button type="button" class="hotel-item__more-btn">Подробнее</button>
    </div>
  </div>
  <div class="hotel-item__right">
    <div class="hotel-item__stars hotel-item__stars_one"></div>
    <div class="hotel-item__rating">Рейтинг: 10</div>
  </div>
</article>
`;
